e2e/__tests__/circularInequality.test.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {tmpdir} from 'os';
+import * as path from 'path';
+import {wrap} from 'jest-snapshot-serializer-raw';
+import {
+  cleanup,
+  createEmptyPackage,
+  extractSortedSummary,
+  writeFiles,
+} from '../Utils';
+import {runContinuous} from '../runJest';
+
+const tempDir = path.resolve(tmpdir(), 'circular-inequality-test');
+
+beforeEach(() => {
+  createEmptyPackage(tempDir);
+});
+
+afterEach(() => {
+  cleanup(tempDir);
+});
+
+test('handles circular inequality properly', async () => {
+  const testFileContent = `
+    it('test', () => {
+      const foo = {};
+      foo.ref = foo;
+
+      expect(foo).toEqual({});
+    });
+
+    it('test 2', () => {
+      const foo = {};
+      foo.ref = foo;
+
+      expect(foo).toEqual({});
+    });
```

Comment:
Is there anything different between the two tests that I'm overlooking or are they the same?

Reply:
They're identical. Might be enough to just have one, haven't verified

Reply:
not needed, so removed it 👍